lib/collection/src/shards/forward_proxy_shard.rs

```diff
@@ -390,6 +402,35 @@ impl ShardOperation for ForwardProxyShard {
 
         let _update_lock = self.update_lock.lock().await;
 
+        let points_matching_filter_before = {
+            if let Some(filter) = &self.filter {
+                if let Some(point_ids) = operation.operation.point_ids() {
+                    let filter = filter.clone();
+
+                    self.wrapped_shard
+                        .scroll_by(
+                            None,
+                            point_ids.len(),
+                            &WithPayloadInterface::Bool(false),
+                            &WithVector::Bool(false),
+                            Some(&filter.with_point_ids(point_ids)),
+                            &Handle::current(),
+                            None,
+                            None,                           // No timeout
+                            HwMeasurementAcc::disposable(), // Internal operation, no need to measure hardware here?
+                        )
+                        .await?
+                        .into_iter()
+                        .map(|record| record.id)
+                        .collect::<HashSet<_>>()
+                } else {
+                    HashSet::new()
+                }
+            } else {
+                HashSet::new()
+            }
+        };
```

Comment:
Is this necessary?

I believe we we agreed in a daily that we don't want to read all points first to decide on whether we forward the operation.

Is it possible to just forward everything, or just the operations that also target the other shard key?

Or is this not possible requiring us to do this?